### PR TITLE
RC65.1 no crash on X

### PR DESCRIPTION
**Pull request by AndrewMeadows**
- re-order cleanup during shutdown

- disable LogHandler at some point during shutdown

### CODE REVIEW

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -2273,9 +2277,6 @@ void Application::cleanupBeforeQuit() {
     // Clear any queued processing (I/O, FBX/OBJ/Texture parsing)
     QThreadPool::globalInstance()->clear();
 
-    DependencyManager::get<ScriptEngines>()->shutdownScripting(); // stop all currently running global scripts
-    DependencyManager::destroy<ScriptEngines>();
-
```
Comment (AndrewMeadows) [inline interface/src/Application.cpp]: We need to shutdown scripts earlier in `Application::cleanupBeforeQuit()`.

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -2312,9 +2313,8 @@ void Application::cleanupBeforeQuit() {
     // stop QML
     DependencyManager::destroy<TabletScriptingInterface>();
     DependencyManager::destroy<ToolbarScriptingInterface>();
-    DependencyManager::destroy<OffscreenUi>();
-
     DependencyManager::destroy<OffscreenQmlSurfaceCache>();
+    DependencyManager::destroy<OffscreenUi>();
```
Comment (AndrewMeadows) [inline interface/src/Application.cpp]: Cleanup `OffscreenQmlSurfaceCache` before `OffscreenUi`, in reverse order from how they are created.

File: interface/src/Application.cpp [hunk 3]
```diff
@@ -2405,6 +2405,8 @@ void Application::teardown() {
 
     // Can't log to file passed this point, FileLogger about to be deleted
     qInstallMessageHandler(LogHandler::verboseMessageHandler);
+    // HACK: shutdown LogHandler to avoid tangles as multiple threads collapse
+    LogHandler::shutdown();
```
Comment (AndrewMeadows) [inline interface/src/Application.cpp]: After fixing the `OffscreenQMLSurfaceCache` issue I was experiencing a deadlock on shutdown: trying to delete the `LogHandler` while some other context was trying to log a message.  Completely disabling the `LogHandler` here fixes the problem, but this doesn't seem like the proper fix which is why I labeled it as a `HACK`.

**Comment by AndrewMeadows** [comment]
The **repro recipe** that was working for me on Windows:

**(1)** Login to "SteamedHams" with your mouse hovering over the **[X]** close button in the corner of the window.
**(2)** As soon as the scene starts to load press the **[X]** button to logout.
--> in RC65.1: crash on shutdown sometimes (1 in 5 times)
--> in this PR: does not crash on shutdown (0 in 11 times)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8164/app_12688_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8164/HighFidelity-Beta-PR12688-d83da35c43844faec260c99053824474f89bae54.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8164/HighFidelity-Beta-PR12688-d83da35c43844faec260c99053824474f89bae54.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
On Mac and Windows, the interface never crashed while closing down via X. Each system was tested 15 times. 👍

**Comment by conklin94122** [comment]
Thank you @qlimnavarro .

I believe there are a few other issues that need to be worked out in this PR, so I'm not going to QA Approve it yet. QA should expect this to come back around as QA Ready again.

**Comment by conklin94122** [comment]
Removing QA Ready as I believe @AndrewMeadows has additional changes planned that would need to be retested.

**Pull request closed without merging**